### PR TITLE
Initial Commit of SNMP Plugin

### PR DESCRIPTION
**Pull request by sisamiwe**
Hi,

this is the 2nd try to create PR for SNMP.
Michael

### CODE REVIEW

**Comment by psilo909** [comment]
i dont think these two are required?
A snmp/__pycache__/__init__.py.1 (148)
A snmp/__pycache__/__init__.py.2 (135)

File: snmp/__init__.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+        Initalizes the plugin. The parameters describe for this method are pulled from the entry in plugin.conf.
+        """
+
+        from bin.smarthome import VERSION
+        if '.'.join(VERSION.split('.', 2)[:2]) <= '1.5':
+            self.logger = logging.getLogger(__name__)
+
+        if self.logger.isEnabledFor(logging.DEBUG):
+            self.logger.debug("init {}".format(__name__))
+        self._sh = self.get_sh()
```
Comment (psilo909) [inline snmp/__init__.py]: do you really need that line?

Comment (sisamiwe) [reply snmp/__init__.py]: which line your are talking about?

**Comment by sisamiwe** [comment]
@psilo909 

the 3 files have been deleted from commit

**Comment by sisamiwe** [comment]
Hey,
is there anything more to do from my side in regards to that PR?

**Comment by psilo909** [comment]
There are more comments from my side above. Did u address them? E.g. minversion

File: snmp/__init__.py (hunk 1 above)
Comment (msinn) [reply snmp/__init__.py]: self.logger.debug(...) does the trick of logging only if debug logging is enabled. self.logger.isEnabledFor(logging.DEBUG) has the same effect so it is redundant.

Comment (sisamiwe) [reply snmp/__init__.py]: OK, I will remove that and do a new PR.
Are there more thing to do?

**Comment by sisamiwe** [comment]
@psilo909 
Where have you made these comments you are talking about?

**Comment by psilo909** [comment]
just above this post?

File: snmp/plugin.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    # Global plugin attributes
+    type: protocol                  # plugin type (gateway, interface, protocol, system, web)
+    description:
+        de: 'SNMP Queries'
+        en: 'SNMP Queries'
+    maintainer: Michael Wenzel
+    state: develop                 # change to ready when done with development
+
+    version: 1.6.0                 # Plugin version
+    sh_minversion: 1.4.2           # minimum shNG version to use this plugin
```
[the remark this answers is not in the document]
Comment (psilo909) [reply snmp/plugin.yaml]: sorry it looks like i should have submitted my review :-/

[the remark this answers is not in the document]
Comment (sisamiwe) [reply snmp/plugin.yaml]: That's why i asked for your comments.
I've done all that and try to do a new commit later on.

File: snmp/__init__.py [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+            # data['item'] = {}
+            # for i in self.plugin.items:
+            #     data['item'][i]['value'] = self.plugin.getitemvalue(i)
+            #
+            # return it as json the the web page
+            # try:
+            #     return json.dumps(data)
+            # except Exception as e:
+            #     self.logger.error("get_data_html exception: {}".format(e))
+        return {}
```
[the remark this answers is not in the document]
Comment (sisamiwe) [reply snmp/__init__.py]: newline added in new commit

File: snmp/plugin.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
+# Metadata for the Smart-Plugin
+plugin:
```
[the remark this answers is not in the document]
Comment (sisamiwe) [reply snmp/plugin.yaml]: # Support Thread link added in new commit
# self._sh = self.get_sh() deleted in new commit

File: snmp/plugin.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (sisamiwe) [reply snmp/plugin.yaml]: I just tested it wir 1.6.x the the min version is set to 1.6.0 in new commit

**Pull request closed without merging**